createtransfers/createtransfers.py

```diff
@@ -0,0 +1,413 @@
+                u'\u265A\u265B\u265C\u265D\u265E'
+                u'\u265F'))
+    }
+)
+
+
+def create_file_and_write(file_name, file_path, encoding):
+    """Given a filename and encoding write a file with self descriptive
+    content to disk.
+    """
+    file_name_unicode = file_name
+    file_name_bytes = file_name_unicode.encode(encoding)
+    file_path_bytes = os.path.join(file_path, file_name_bytes)
+    try:
+        with open(file_path_bytes, 'w') as fout:
+            fout.write('The name of this file is encoded with encoding'
+                       ' {}\n'.format(encoding))
+            LOGGER.info("Created: %s", file_path_bytes)
+    except (OSError, IOError) as err:
+        ret = "Failed to open file: {0}, error: {1}".format(file_path_bytes,
+                                                            err)
+        raise CreateTransferException(ret)
+
+
+def rm_dirs_and_create(dir_path):
+    """Remove pre-existing director structures to write new ones out
+    to disk.
+    """
+    if os.path.exists(dir_path):
+        try:
+            shutil.rmtree(dir_path)
+        except OSError as err:
+            ret = "Failed to remove dirtree: {0} {1}".format(dir_path, err)
+            raise CreateTransferException(ret)
+    try:
+        os.makedirs(dir_path)
+    except (os.error) as err:
+        ret = "Failed to make directory: {0} {1}".format(dir_path, err)
+        raise CreateTransferException(ret)
+
+
+def create_transfer_collection(collection_path, media_path,
+                               media_n, new_media_dirname,
+                               media_ext):
+    """To create our transfers we are copying from elsewhere within the
+    sample data repository to generate a new collection.
+    """
+
+    # Stat file, and return asap if it doesn't exist. If it does exist,
+    # provide the user an idea about the size of the collection being
+    # generated.
+    try:
+        stat_file(media_path, media_n)
+        new_media_dir = os.path.join(collection_path, new_media_dirname)
+    except CreateTransferException as err:
+        LOGGER.error(err)
+        return
+
+    try:
+        rm_dirs_and_create(new_media_dir)
+    except CreateTransferException as err:
+        LOGGER.error(err)
+        return
+
+    copy_media_location(media_path, new_media_dir, media_n, media_ext)
+
+
+def copy_media_location(copy_media_path, new_media_dir, media_n, media_ext):
+    """Create as maany copies of our sampledata files as required by the
+    user in the argument media_n.
+    """
+    for i in range(media_n):
+        new_media_fname = 'mediafile_{0}.{1}'.format(i, media_ext)
+        new_media_path = os.path.join(new_media_dir, new_media_fname)
+        shutil.copyfile(copy_media_path, new_media_path)
+    LOGGER.info("%s files created at %s", media_n, new_media_dir)
+
+
+def stat_file(file_path, files_n):
+    """ Check that the file exists to make sure we're not referencing objects
```

Comment:
What's with the `""" ` pattern here? I know this is nitpicky, but I would prefer docstrings to be consistent, at least within the module. Could we just consistently use `"""First words of docstring` instead of `""" First words of docstring` or `"""\nFirst words of docstring`?

Reply:
I think this is just an innocent typo...